Failure to open DB after improper shutdown

It was reported [here](https://groups.google.com/forum/#!topic/h2-database/i5PML-CWTZw) that 1.4.199 fails to open db after abrupt shutdown, and recovery tool fails the same way, but 1.4.196 opens it without complains.
```
jambo@jambo8:~$ java -cp h2-1.4.199.jar org.h2.tools.Recover
Exception in thread "main" java.lang.IllegalStateException: Reading from nio:/home/jambo/database.mv.db failed; file length 1127067648 read length 4096 at 1152263673 [1.4.199/1]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:420)
	at org.h2.mvstore.FileStore.readFully(FileStore.java:98)
	at org.h2.mvstore.MVStore.readBufferForPage(MVStore.java:1048)
	at org.h2.mvstore.MVStore.readPage(MVStore.java:2186)
	at org.h2.mvstore.MVMap.readPage(MVMap.java:554)
	at org.h2.mvstore.Page$NonLeaf.getChildPage(Page.java:1086)
	at org.h2.mvstore.Cursor.hasNext(Cursor.java:53)
	at org.h2.mvstore.MVStore.readStoreHeader(MVStore.java:793)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:401)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343)
	at org.h2.mvstore.MVStoreTool.info(MVStoreTool.java:347)
	at org.h2.tools.Recover.process(Recover.java:344)
	at org.h2.tools.Recover.runTool(Recover.java:195)
	at org.h2.tools.Recover.main(Recover.java:158)
Caused by: java.io.EOFException
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:408)
```

--- a/h2/src/main/org/h2/mvstore/MVStore.java
+++ b/h2/src/main/org/h2/mvstore/MVStore.java
@@ -831,33 +831,37 @@ public class MVStore implements AutoCloseable {
         boolean verified = false;
         while(!verified && setLastChunk(lastChunkCandidates.poll()) != null) {
             verified = true;
-            // load the chunk metadata: although meta's root page resides in the lastChunk,
-            // traversing meta map might recursively load another chunk(s)
-            Cursor<String, String> cursor = meta.cursor("chunk.");
-            while (cursor.hasNext() && cursor.next().startsWith("chunk.")) {
-                Chunk c = Chunk.fromString(cursor.getValue());
-                assert c.version <= currentVersion;
-                // might be there already, due to meta traversal
-                // see readPage() ... getChunkIfFound()
-                chunks.putIfAbsent(c.id, c);
-                long block = c.block;
-                test = validChunkCacheByLocation.get(block);
-                if (test == null) {
-                    test = readChunkHeaderAndFooter(block);
-                    if (test != null && test.id == c.id) { // chunk is valid
-                        validChunkCacheByLocation.put(block, test);
-                        lastChunkCandidates.offer(test);
+            try {
+                // load the chunk metadata: although meta's root page resides in the lastChunk,
+                // traversing meta map might recursively load another chunk(s)
+                Cursor<String, String> cursor = meta.cursor("chunk.");
+                while (cursor.hasNext() && cursor.next().startsWith("chunk.")) {
+                    Chunk c = Chunk.fromString(cursor.getValue());
+                    assert c.version <= currentVersion;
+                    // might be there already, due to meta traversal
+                    // see readPage() ... getChunkIfFound()
+                    chunks.putIfAbsent(c.id, c);
+                    long block = c.block;
+                    test = validChunkCacheByLocation.get(block);
+                    if (test == null) {
+                        test = readChunkHeaderAndFooter(block);
+                        if (test != null && test.id == c.id) { // chunk is valid
+                            validChunkCacheByLocation.put(block, test);
+                            lastChunkCandidates.offer(test);
+                            continue;
+                        }
+                    } else if (test.id == c.id) { // chunk is valid
+                        // nothing to do, since chunk was already verified
+                        // and registered as potential "last chunk" candidate
                         continue;
                     }
-                } else if (test.id == c.id) { // chunk is valid
-                    // nothing to do, since chunk was already verified
-                    // and registered as potential "last chunk" candidate
-                    continue;
+                    // chunk reference is invalid
+                    // this "last chunk" candidate is not suitable
+                    // but we continue to process all references
+                    // to find other potential candidates
+                    verified = false;
                 }
-                // chunk reference is invalid
-                // this "last chunk" candidate is not suitable
-                // but we continue to process all references
-                // to find other potential candidates
+            } catch(IllegalStateException ignored) {
                 verified = false;
             }
         }

